dd-sdk-android/src/main/kotlin/com/datadog/android/rum/gestures/DatadogGesturesListener.kt

```diff
@@ -0,0 +1,155 @@
+package com.datadog.android.rum.gestures
+
+import android.view.GestureDetector
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.tracing.Tracer
+import java.lang.ref.WeakReference
+import java.util.LinkedList
+import java.util.concurrent.TimeUnit
+
+internal class DatadogGesturesListener(
+    private val rumTracer: Tracer,
+    private val decorViewReference: WeakReference<View>
+) :
+    GestureDetector.OnGestureListener {
+
+    // region GesturesListener
+
+    override fun onShowPress(e: MotionEvent) {
+        // No Op
+    }
+
+    override fun onSingleTapUp(e: MotionEvent): Boolean {
+        val decorView: View? = decorViewReference.get()
+        if (decorView != null) {
+            findTarget(decorView, e.x, e.y)?.let { target ->
+                // we just intercept but not steal the event
+                val spanBuilder = rumTracer
+                    .buildSpan(UI_TAP_ACTION_EVENT)
+                spanBuilder.withTag(TAG_TARGET_CLASS_NAME, target.javaClass.canonicalName)
+                // we will handle general exceptions here. We don't know what might go wrong apart from
+                // ResourceNotFound exception.
+                @Suppress("TooGenericExceptionCaught")
+                try {
+                    spanBuilder.withTag(
+                        TAG_TARGET_RESOURCE_ID,
+                        target.resources.getResourceEntryName(target.id)
+                    )
+                } catch (e: Exception) {
```

Comment:
You should only catch  `NotFoundException` I think

Reply:
I was actually not sure here...was trying to play safe.